### PR TITLE
more docs

### PR DESCRIPTION
**Pull request by awni**
## Proposed changes

Mode documentation on MLX usage:

- Lazy evaluation
- Saving and loading
- Indexing

### CODE REVIEW

File: docs/src/usage/indexing.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+         [4, 6]], dtype=int32
+
+You can inex with ``None`` to create a new axis:
+
+.. code-block:: shell
+
+  >>> arr = mx.arange(8)
+  >>> arr.shape
+  [8]
+  >>> arr[None].sape
```
Comment (gboduljak) [inline docs/src/usage/indexing.rst:45]: This is probably a typo. I think we should have ``arr[None].shape`` here.

Comment (awni) [reply docs/src/usage/indexing.rst:45]: probably -> definitely. Thanks for the catch.

File: docs/src/usage/lazy_evaluation.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  for _ in range(100):
+       a = a + b
+       mx.eval(a)
+       b = b * 2
+       mx.eval(b)
+
+This is a bad idea because there is some fixed overhead with each graph
+evaluation. On the other hand, there is some slight overhead which grows with
+the compute graph size, so extremely large graphs (while computationally
+correct), can be also be costly.
```
Comment (angeloskath) [inline docs/src/usage/lazy_evaluation.rst:81]: typo: can --be-- also be

**Review by angeloskath — APPROVED** [review]
This is pure gold! Thanks.

Left a comment for a typo.

**Pull request merged**